src/logger/logger.py

```diff
@@ -21,470 +24,521 @@
+            state_before=event.arguments["state_before"],
+            state_after=event.arguments["state_after"],
         )
 
-    def train_enter_block_section(
-        self,
-        tick: int,
-        train_id: str,
-        block_section_id: str,
-        block_section_length: float,
-    ) -> Type[None]:
+    def train_enter_block_section(self, event: Event) -> Type[None]:
         """
         This function should be called when a train enters a block section. This should include a
         train identifier, the block section identifier and the length of the block section.
-        :param tick: The current simulation tick
-        :param train_id: The id of the train
-        :param block_section_id: The id of the block section
-        :param block_section_length: The length of the block section
-        :rtype: None
+        :param event: the event containing all relevant info
         """
         TrainEnterBlockSectionLogEntry.create(
             timestamp=datetime.now(),
-            tick=tick,
-            message=f"Train with ID {train_id} entered block section with ID {block_section_id} "
-            f"with length {block_section_length}",
+            tick=event.arguments["tick"],
+            message=f"Train with ID {event.arguments['train_id']} entered block "
+            f"section with ID {event.arguments['block_section_id']} "
+            f"with length {event.arguments['block_section_length']}",
             run_id=self.run_id,
-            train_id=train_id,
-            block_section_id=block_section_id,
-            block_section_length=block_section_length,
+            train_id=event.arguments["train_id"],
+            block_section_id=event.arguments["block_section_id"],
+            block_section_length=event.arguments["block_section_length"],
         )
 
-    def train_leave_block_section(
-        self,
-        tick: int,
-        train_id: str,
-        block_section_id: str,
-    ) -> Type[None]:
+    def train_leave_block_section(self, event: Event) -> Type[None]:
         """
         This function should be called when a train leaves a block section. This should include a
         train identifier, the block section identifier and the length of the block section.
-        :param tick: The current simulation tick
-        :param train_id: The id of the train
-        :param block_section_id: The id of the block section
-        :rtype: None
+        :param event: the event containing all relevant info
         """
         TrainLeaveBlockSectionLogEntry.create(
             timestamp=datetime.now(),
-            tick=tick,
-            message=f"Train with ID {train_id} left block section with ID {block_section_id}",
+            tick=event.arguments["tick"],
+            message=f"Train with ID {event.arguments['train_id']} left block "
+            f"section with ID {event.arguments['block_section_id']} "
+            f"with length {event.arguments['block_section_length']}",
```

Comment:
Why did you add to the message?

Reply:
I will change it back. The length is stored in the `LogEntry` anyway.